### PR TITLE
[ansible/artifactory] Bugfix/305

### LINKED ISSUE #305: [ansible/artifactory] File store always configured as HA, excessive logging on single instance
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

The default configuration configures HA in binarystore.xml:

```xml
<config version="2">
    <chain template="cluster-file-system"/>
</config>
```

The configuration for a single server should be:

```xml
<config version="1">
    <chain template="file-system"/>
</config>
```

The installer should count the servers in the group `artifactory_servers` to configure accordingly.

**Which installer**:
Ansible

**Which product and version** (eg: ansible & collection version - 7.24.2):

```
ansible [core 2.14.2]
```
jfrog.platform: 10.13.2

**Which operating system and version**(eg: ubuntu & version - 20.4):
`RHEL8.8`

**Which product license** (Enterprise/Pro):
Enterprise

**JFrog support reference** (if already raised with support team): 

**What happened**:
If there's is only a single instance this leads to excessive logging.

```
Number of active servers in cluster (1) is lower than the redundancy value defined in binarystore.xml. Skipping ping for 'sharding-cluster'
```

**What you expected to happen**:

Proper config, no errors.

**How to reproduce it** (as minimally and precisely as possible):
Just install and look in console log

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by bbaassssiiee**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Title of the PR starts with installer/product name (e.g. `[ansible/artifactory]`)
- [X] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:

Fixes excessive logging on single server Artifactory.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #305 

**Special notes for your reviewer**:

edits binarystore.xml based on server count.

### CODE REVIEW

File: Ansible/ansible_collections/jfrog/platform/roles/artifactory/defaults/main.yml [hunk 1]
```diff
@@ -69,12 +69,21 @@ artifactory_admin_password: password
 artifactory_service_file: /lib/systemd/system/artifactory.service
 
 # Provide binarystore XML content below with 2-space indentation
+# For HA
 artifactory_binarystore: |-
```
Comment (chukka) [inline Ansible/ansible_collections/jfrog/platform/roles/artifactory/defaults/main.yml:73]: @bbaassssiiee Thanks for PR, does it  not make sense to have single name as `artifactory_binarystore` and fix on chain template based on number of servers ? (avoid another variable `artifactory_filestore`)

File: Ansible/ansible_collections/jfrog/platform/roles/artifactory/tasks/main.yml [hunk 2]
```diff
@@ -1,3 +1,14 @@
+- name: Get artifactory_servers count
+  ansible.builtin.set_fact:
+    artifactory_server_count: "{{ groups['artifactory_servers'] | length }}"
```
Comment (chukka) [inline Ansible/ansible_collections/jfrog/platform/roles/artifactory/tasks/main.yml:3]: @bbaassssiiee this logic can be moved to defaults.yaml file

Comment (bbaassssiiee) [reply Ansible/ansible_collections/jfrog/platform/roles/artifactory/tasks/main.yml:3]: I notice a variable `artifactory_ha_enabled` in defaults.yml, looks like a candidate for this logic...

File: Ansible/ansible_collections/jfrog/platform/roles/artifactory/defaults/main.yml (hunk 1 above)
Comment (bbaassssiiee) [reply Ansible/ansible_collections/jfrog/platform/roles/artifactory/defaults/main.yml]: We can select the appropriate XML in the template based on `artifactory_ha_enabled`.

Comment (bbaassssiiee) [reply Ansible/ansible_collections/jfrog/platform/roles/artifactory/defaults/main.yml]: We can select the appropriate XML in the template based on `artifactory_ha_enabled`.

**Comment by chukka** [comment]
looks good , thanks for the changes , will take it in next patch release

**Comment by chukka** [comment]
fixed via #308 FYI - Added an additional change of removal of binary_store variable , closing this

**Pull request closed without merging**